src/completionItemBuilder.ts

```diff
@@ -95,12 +96,22 @@ export class CompletionItemBuilder {
   private replaceExpression = (replacement: string, code: string, customRegex?: string) => {
     const re = new RegExp(customRegex || RegexExpression, 'g')
 
-    return replacement.replace(re, (_match, p1) => {
-      if (p1 && this.filters[p1]) {
-        return this.filters[p1](code)
+    return replacement.replace(re, (_match, type, variant) => {
+      let codeToInsert: string;
+      if (type === 'expr') {
+        codeToInsert = code;
+      } else if (type === 'exprLast') {
+        codeToInsert = getLastExpressionName(this.node);
+      } else {
```

Comment:
Can you make this explicit instead of an `else`? Perhaps when there is no match we should throw exception. It should never happen unless you forget to implement one of the options but since there is no strong relation between this and the regex it's impossible to be sure.

Reply:
Makes sense, but I think I'll be rewriting it a bit to support groups from regexs as we discussed in https://github.com/ipatalas/vscode-postfix-ts/issues/90#issuecomment-1295357513